### PR TITLE
Add support for generating SVM relocations on Power

### PR DESCRIPTION
**Pull request by aviansie-ben**
This PR fixes a number of problematic sections of the Power codegen that did not properly handle being run under AOT. While these sections were previously not run under AOT due to the limitations of AOT in general, they _can_ run when SVM AOT is being used. These sections have been fixed up to properly handle adding AOT relocations when run under SVM AOT.

### CODE REVIEW

**Comment by dsouzai** [comment]
@gita-omr @IBMJimmyk could you guys please review?

File: runtime/compiler/p/codegen/CallSnippet.cpp [hunk 1]
```diff
@@ -282,9 +282,13 @@ TR_RuntimeHelper TR::PPCCallSnippet::getInterpretedDispatchHelper(
       isJitInduceOSRCall = true;
       }
 
-   if (methodSymRef->isUnresolved() || comp->compileRelocatableCode()) // KEN
+   bool forceUnresolvedDispatch = fej9->forceUnresolvedDispatch();
```
[the remark this answers is not in the document]
Comment (dsouzai) [reply runtime/compiler/p/codegen/CallSnippet.cpp]: Yeah, `forceUnresolvedDispatch` returns `true` under AOT.

[the remark this answers is not in the document]
Comment (aviansie-ben) [reply runtime/compiler/p/codegen/CallSnippet.cpp]: Yes, as far as I know they're just two different names for the same query.

File: runtime/compiler/p/codegen/CallSnippet.cpp [hunk 2]
```diff
@@ -414,10 +422,18 @@ uint8_t *TR::PPCCallSnippet::emitSnippetBody()
          if (comp->getOption(TR_EnableHCR))
             cg()->jitAddPicToPatchOnClassRedefinition((void *)methodSymbol->getMethodAddress(), (void *)cursor);
 
-         cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor, (uint8_t *)methodSymRef,
-                                                                                 getNode() ? (uint8_t *)getNode()->getInlinedSiteIndex() : (uint8_t *)-1,
-                                                                                 TR_MethodObject, cg()),
-                                      __FILE__, __LINE__, callNode);
+         if (comp->getOption(TR_UseSymbolValidationManager))
```
[the remark this answers is not in the document]
Comment (aviansie-ben) [reply runtime/compiler/p/codegen/CallSnippet.cpp]: This code can never be reached under AOT if SVM is disabled, as `forceUnresolvedDispatch` will always be true in that case.

File: runtime/compiler/p/codegen/CallSnippet.cpp [hunk 3]
```diff
@@ -282,9 +282,13 @@ TR_RuntimeHelper TR::PPCCallSnippet::getInterpretedDispatchHelper(
       isJitInduceOSRCall = true;
       }
 
-   if (methodSymRef->isUnresolved() || comp->compileRelocatableCode()) // KEN
+   bool forceUnresolvedDispatch = fej9->forceUnresolvedDispatch();
+   if (comp->getOption(TR_UseSymbolValidationManager))
+      forceUnresolvedDispatch = false;
+
+   if (methodSymRef->isUnresolved() || forceUnresolvedDispatch) // KEN
```
[the remark this answers is not in the document]
Comment (aviansie-ben) [reply runtime/compiler/p/codegen/CallSnippet.cpp]: Agreed. I'll go ahead and remove that.

File: runtime/compiler/p/codegen/CallSnippet.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (gita-omr) [reply runtime/compiler/p/codegen/CallSnippet.cpp]: Might be more clear if we check 'is AOT Compilation' vs. 'Is SVM'.

[the remark this answers is not in the document]
Comment (aviansie-ben) [reply runtime/compiler/p/codegen/CallSnippet.cpp]: Alright, I'll go ahead and change that

**Comment by aviansie-ben** [comment]
I've clarified a bit about the object allocation inlining changes in the description of the relevant commit. All other issues brought up have also been resolved.

**Comment by gita-omr** [comment]
Jenkins test sanity,extended aix,plinux jdk8,jdk11

**Comment by dsouzai** [comment]
Jenkins may not be available until sometime tomorrow afternoon (CET) because of maintenance.

**Comment by dsouzai** [comment]
Jenkins test sanity,extended aix,plinux jdk8,jdk11

**Comment by dsouzai** [comment]
Jenkins test sanity aix,plinux jdk8,jdk11

**Comment by dsouzai** [comment]
Jenkins test extended aix,plinux jdk8,jdk11

**Comment by dsouzai** [comment]
extended jdk11 inux_ppc-64_cmprssptrs_le build failed with
```
17:08:04  + make compile
17:08:04  settings.mk:46: /home/jenkins/jenkins-agent/workspace/PullRequest-Extended-JDK11-linux_ppc-64_cmprssptrs_le-OpenJ9/openj9/test/TestConfig/../TestConfig/utils.mk: No such file or directory
17:08:04  makefile:39: count.mk: No such file or directory
17:08:04  make: *** No rule to make target 'count.mk'.  Stop.
```

**Comment by dsouzai** [comment]
Jenkins test extended plinux jdk11

**Comment by gita-omr** [comment]
Looks like Jenkins is still not working?

**Comment by gita-omr** [comment]
Jenkins test sanity,extended aix,plinux jdk8,jdk11

**Comment by gita-omr** [comment]
Jenkins test sanity plinux jdk8,jdk11

**Comment by aviansie-ben** [comment]
Looks like the extended test AIX failure is a known issue unrelated to these changes: #3081
As for the extended PPC64LE test failure, it's also been noted as an infra issue that should be corrected now: #4830

**Pull request merged**